### PR TITLE
Closes #2316: Adds BrowserMenuCaption

### LINKED ISSUE #2316: Create a caption menu item
For Fenix we will need a caption menu item in the browser fragment when a custom tab is being used. We should create a menu item for this with a customizable text size for future flexibility.

[See mocks here](https://github.com/mozilla-mobile/fenix/issues/203#issuecomment-463426620)

### PR DESCRIPTION
**Pull request by sblatz**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

File: components/browser/menu/src/test/java/mozilla/components/browser/menu/item/BrowserMenuCaptionTest.kt [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import org.junit.Test
+
+class BrowserMenuCaptionTest {
+
+    @Test
+    fun `browser menu caption uses correct layout`() {
+        val item = BrowserMenuCaption("Powered by Mozilla")
+        assertEquals(R.layout.mozac_browser_menu_item_caption, item.getLayoutResource())
+    }
+}
```
Comment (pocmo) [inline components/browser/menu/src/test/java/mozilla/components/browser/menu/item/BrowserMenuCaptionTest.kt]: Maybe a test to check the text is actually set? :)

Comment (sblatz) [reply components/browser/menu/src/test/java/mozilla/components/browser/menu/item/BrowserMenuCaptionTest.kt]: Ah, I didn't think there was a way to test this, but I looked at `BrowserMenuImageTextTest` for some inspiration. I've added a few more, but please let me know if that's not sufficient!

**Comment by Amejia481** [comment]
@sblatz what is the difference between `BrowserMenuCaption` and `SimpleBrowserMenuItem`?
it's only the text size and the no click listener attach to it, can we add it SimpleBrowserMenuItem`?

**Comment by sblatz** [comment]
@Amejia481 Yes, those are the only differences. I'm happy to add a `textSize` param to `SimpleBrowserMenuItem`, but I also didn't want tapping on it to dismiss the menu, which is why I ended up creating a separate item. Is there another way I can get around this?

**Comment by Amejia481** [comment]
@sblatz we could make, the listener optional and add set the click listener when is not null
 
```kotlin
class SimpleBrowserMenuItem(
        private val label: String,
        @ColorRes
        private val textColorResource: Int = NO_ID,
        private val listener: (() -> Unit)? = null
) : BrowserMenuItem {
    override var visible: () -> Boolean = { true }

    override fun getLayoutResource() = R.layout.mozac_browser_menu_item_simple

    override fun bind(menu: BrowserMenu, view: View) {
        (view as TextView).text = label
        if (textColorResource != NO_ID) {
            view.setTextColor(ContextCompat.getColor(view.context, textColorResource))
        }

        if (listener != null) {
            view.setOnClickListener {
                listener.invoke()
                menu.dismiss()
            }
        } else {
            // This will remove the ripple effect.
            view.background = null
        }
    }
}
```

**Comment by sblatz** [comment]
@Amejia481 seems like a great solution. I'll make that change!

File: components/browser/menu/src/main/res/layout/mozac_browser_menu_item_caption.xml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (Amejia481) [inline components/browser/menu/src/main/res/layout/mozac_browser_menu_item_caption.xml]: With the new change, I think we are not going to need this.

Comment (sblatz) [reply components/browser/menu/src/main/res/layout/mozac_browser_menu_item_caption.xml]: Sorry about that, went to lunch and hadn't finished up my changes. Should be ready for review now!

Comment (Amejia481) [reply components/browser/menu/src/main/res/layout/mozac_browser_menu_item_caption.xml]: Don't worry :)

**Pull request merged**